### ISSUE TITLE
Modal component needs to trap keyboard focus when open

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#214]
The [Modal component](https://sparkpost.github.io/matchbox/?selectedKind=Overlays%7CModal&selectedStory=Toggle%20Example&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) needs to trap focus when open.

## Requirements

* When a modal is open, focus should be moved to the modal element itself programmatically - this can be achieved usually by using a `ref` (example [here](https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element)) and adding `tabIndex="-1"`
* As the user tabs through focusable elements within the modal instance, the user should be unable to 'leave' the modal via keyboard tabbing
* This also applies to `shift+tab` when the user is on the first focusable element within a modal instance. In this situation, focus should move to the last focusable element within a modal instance.

## Resources

* The [Dialog Component in Reach UI](https://ui.reach.tech/dialog)
* Potential easy fix by leveraging [React Focus Lock](https://www.npmjs.com/package/react-focus-lock)
